crates/biome_formatter/src/format_element/document.rs

```diff
@@ -576,6 +576,38 @@ impl FormatElements for [FormatElement] {
         false
     }
 
+    fn may_directly_break(&self) -> bool {
+        use Tag::*;
+        let mut ignore_depth = 0usize;
+
+        for element in self {
+            match element {
+                // Line suffix
+                // Ignore if any of its content breaks
+                FormatElement::Tag(StartLineSuffix) => {
+                    ignore_depth += 1;
+                }
+                FormatElement::Tag(EndLineSuffix) => {
+                    ignore_depth -= 1;
+                }
+                FormatElement::Interned(interned) if ignore_depth == 0 => {
+                    if interned.may_directly_break() {
+                        return true;
+                    }
+                }
```

Comment:
I'd like to not duplicate this code again, but idk a cleaner way to write it out. I _think_ this state management is still necessary for this function, since it also shouldn't be inspecting text content on suffixes, but I'm not 100% sure.